docs/api-reference-1/frontity.md

```diff
@@ -648,3 +651,207 @@ const decodedText = decode("milk &amp; cookies");
 ```
 
+### `Slot`
+
+The `<Slot />` component enables the use of a powerful pattern called Slot and Fill.
+This allows for  any React component to be inserted into, or hooked onto, different places within the app, thereby improving extensibility.
+
+This component allows a theme developer to insert named `<Slot>` components in various places in a theme.
+Other package developers are then able to add *'fill'* components which will be hooked onto the named slots.
+
+#### Rationale
+
+When developing a site the developer is often required to make certain customisations to the structure and/or appearance of the site.
+This can be difficult to do and necessitates modifying the core code of the theme.
+
+Theme developers are able to facilitate such customisations by adding `<Slot />` components at various places in the theme, e.g. above the header, below the header, before the content, etc...
+
+These 'slots' can then be filled with custom components that have been added by the site developer and which are then 'hooked' onto a particular 'slot' to insert the content in that place on the page.
+
+An example might be as follows - the site developer wants to place a third party ad above the content of each page. The theme developer has thoughtfully provided a slot in that position in the theme:
+
+```tsx
+//...
+const Content = () => {
+  //...
+  <Container>
+  <Slot name="Before Content">
+  //...
+  </Container>
+  //...
+}
+```
+
+The site developer is now able to 'hook' a component that returns an ad onto that slot, so that the ad gets rendered in that position on the page. This component is referred to as a *'fill'*.
+
+#### Syntax
+
+```tsx
+<Slot name="name of the slot" data={data} myprops={myprops} />
+```
+or
+```tsx
+<Slot name="name of the slot" data={data} myprops={myprops}>
+  {children}
+</Slot>
+```
+
+#### Props
+
+| Name             | Type      | Default     | Required | Description                                                                                                                                                           |
+| :--------------- | :-------- | :---------- | :------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `name`           | string    | `undefined` | yes     | The name of the Slot. The user of this Slot will have to specify this name in order to insert a Fill component.                                                       |
+| `children`       | ReactNode | `undefined` | yes     | The component that will be used as a fallback in case that no fill is specified for a particular Slot. You can use any type of data that is valid as a react element. |
+| `data`           | any       | `undefined` | no    | Any data that you might want to pass to the Fill. Normally used for passing route data fetched in the parent component.                                               |
```

Comment:
Maybe we could be more specific and mention that `data` should be a data object, like the ones you get when you use `state.source.get()`.

Reply:
I think that Luis meant that the data can have this type: https://github.com/frontity/frontity/blob/dev/packages/source/types/data.ts#L14 because if the user does not pass a value to `data` we call `state.source.get()` automatically which returns this `Data` object. I've updated this now in the description🙂

Reply:
Yes, that's what I meant. Thanks @michalczaplinski and thanks for the update! 👍